src/install/extract_tarball.zig

```diff
@@ -120,7 +120,17 @@ fn extract(this: *const ExtractTarball, log: *logger.Log, tgz_bytes: []const u8)
     };
     const basename = brk: {
         var tmp = name;
-        if (tmp[0] == '@') {
+
+        // Handle URLs - extract just the filename from the URL
+        if (strings.hasPrefixComptime(tmp, "https://") or strings.hasPrefixComptime(tmp, "http://")) {
+            tmp = std.fs.path.basename(tmp);
+            // Remove .tgz or .tar.gz extension if present
+            if (strings.endsWithComptime(tmp, ".tgz")) {
+                tmp = tmp[0 .. tmp.len - 4];
+            } else if (strings.endsWithComptime(tmp, ".tar.gz")) {
+                tmp = tmp[0 .. tmp.len - 7];
+            }
```

Comment:
```suggestion
            if (strings.endsWithComptime(tmp, ".tgz")) {
                tmp = tmp[0 .. tmp.len - ".tgz".len];
            } else if (strings.endsWithComptime(tmp, ".tar.gz")) {
                tmp = tmp[0 .. tmp.len - ".tar.gz".len];
            }
```

better to be explicit about the length's reason - they both should be compiled to the same thing anyway